dev/README_AIRFLOW3_DEV.md

```diff
@@ -0,0 +1,152 @@
+  - [Developing for Airflow 2.11](#developing-for-airflow-211)
+- [Committers / PMCs](#committers--pmcs)
+  - [Merging PRs for providers and Helm chart](#merging-prs-for-providers-and-helm-chart)
+  - [Merging PR for Airflow 3 and 2.10.x / 2.11.x](#merging-pr-for-airflow-3-and-210x--211x)
+  - [Merging PRs 2.10.x](#merging-prs-210x)
+  - [Merging PRs for Airflow 3](#merging-prs-for-airflow-3)
+  - [Merging PRs for Airflow 2.11](#merging-prs-for-airflow-211)
+- [Milestones for PR](#milestones-for-pr)
+  - [Set 2.10.x milestone](#set-210x-milestone)
+  - [Set 2.11 milestone](#set-211-milestone)
+  - [Set 3 milestone](#set-3-milestone)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Main branch is Airflow 3
+
+The main branch is for development of Airflow 3.
+Airflow 2.10.x releases will be cut from `v2-10-stable` branch.
+
+# Contributors
+
+The following section explains which branches you should target your PR to
+
+## Developing for providers and Helm chart
+
+PRs should target `main` branch.
+Make sure your code is only about Providers or Helm chart.
+Avoid mixing core changes into the same PR
+
+## Developing for Airflow 3 and 2.10.x / 2.11.x
+
+If the PR is relevant for both Airflow 3 and 2 it should target `main` branch.
+
+Note: The mental model of Airflow 2.11 is bridge release for Airflow 3.
+As a result, Airflow 2.11 is not planned to introduce new features other than ones relevant as bridge release for Airflow 3.
+That said, we recognize that there may be exceptions.
+If you believe a specific feature is must have for Airflow 2.11 you need to raise this as discussion thread in the mailing list.
+Points to address to make your case:
+
+1. You must clarify what is the urgency (why it can't wait for Airflow 3)
+2. You need be willing to deliver the feature for both main branch and Airflow 2.11 branch
+3. You must be willing to provide support future bug fixes as needed.
+
+Points to consider on how PMC members evaluate the request of exception:
+
+1. Feature impact. Is it really urgent? How many are affected?
+2. Are there any workarounds?
+3. Scope of change. Both in code lines / number of files and components changed.
+4. Is the feature at the heart of Airflow (scheduler, dag parser) or peripheral.
+5. Identity of the requester. Is the request from/supported by a member of the community?
+6. Similar previous cases approved.
+7. Other considerations that may raise by PMC members depending on the case.
+
+## Developing for Airflow 3
+
+PRs should target `main` branch.
+
+## Developing for Airflow 2.10.x
+
+PR should target `v2-10-test` branch. When cutting a new release for 2.10 release manager
+will sync `v2-10-test`  branch to `v2-10-stable` and cut the release from the stable branch.
+PR should never target `v2-10-stable` unless specifically instructed by release manager.
+
+## Developing for Airflow 2.11
+
+Version 2.11 is planned to be cut from `v2-10-stable` branch.
```

Comment:
To we need to put PRs for 2.11 "on hold" so that they are not mixed-up with other patches while we are maintaining 2.10 (and not ready for all stuff being ready in 2.11)?

Reply:
My proposal would be to have a v2-11-test branch where they will get merged, intitially cut-off from v2-10-test, and rebasing that branch when we release patchlevel versions of 2.10.0 (I.e. rebase v2-11-test on top of v2-10-test after every patchlevel is released.